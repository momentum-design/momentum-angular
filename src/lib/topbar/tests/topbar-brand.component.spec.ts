import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { TopbarBrandComponent } from '../topbar-brand.component';

describe('TopbarBrandComponent', () => {
  let testComponent: TopbarBrandComponent;
  let fixture: ComponentFixture<TopbarBrandComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TopbarBrandComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TopbarBrandComponent);
    testComponent = fixture.componentInstance;
  });

  it('should create and match snapshot', () => {
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should handle icon', () => {
    testComponent.icon = 'handset_24';
    fixture.detectChanges();
    const logo = fixture.nativeElement.querySelector('.md-brand__logo');
    const icon = logo.querySelector('i');
    expect(icon.className).toContain('handset_24');
  });

  it('should handle image', () => {
    testComponent.image = 'test.png';
    fixture.detectChanges();
    const logo = fixture.nativeElement.querySelector('.md-brand__logo');
    const img = logo.querySelector('img');
    expect(img.getAttribute('src')).toEqual('test.png');
  });

  it('should handle image alt', () => {
    testComponent.image = 'test.png';
    testComponent.alt = 'test';
    fixture.detectChanges();
    const logo = fixture.nativeElement.querySelector('.md-brand__logo');
    const img = logo.querySelector('img');
    expect(img.getAttribute('alt')).toEqual('test');
  });

  it('should handle title', () => {
    testComponent.title = 'Momentum UI';
    fixture.detectChanges();
    const title = fixture.nativeElement.querySelector('.md-brand__title');
    expect(title.textContent).toEqual('Momentum UI');
  });
});
