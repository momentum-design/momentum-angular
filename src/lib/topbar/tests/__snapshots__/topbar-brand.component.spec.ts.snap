// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarBrandComponent should create and match snapshot 1`] = `
<div
  icon={[Function String]}
>
  <div
    class="md-brand__logo"
    ng-reflect-klass="md-brand__logo"
    ng-reflect-ng-class=""
  >
    
    
    <i
      class="icon icon-cisco-logo"
      ng-reflect-ng-class="icon,icon-cisco-logo"
    />
  </div><div
    class="md-brand__title"
  >
    
  </div>
</div>
`;
